src/java/org/apache/cassandra/utils/CassandraVersion.java

```diff
@@ -200,6 +200,11 @@ private static Integer tryParseInt(String str)
         }
     }
 
+    public boolean isLowerThan(int major, int minor)
```

Comment:
ok

Reply:
Actually, I'd not remove it - the functionality it provides is not covered by simple version comparison because, for example:
`4.0.0-SNAPSHOT < 4.0.0`

Is there something which can be used as a lower bound for 4.0 family which does not look confusing?

This mechanism is intended to detect minor version upgrades so we do not care what is going on in patch, pre-release or build component of the version because we should not have to change the behaviour of the node in such situation

Reply:
A 4.0 lower bound version constant can be defined, just as the one defined in `SystemKeyspace#CURRENT_VERSION`. The earliest canonical 4.0 version is `4.0-alpha1`, which can be used as the lower bound. 

```java
        v1 = new CassandraVersion("4.0-alpha1");
        v2 = new CassandraVersion("4.0.0-snapshot");
        v3 = new CassandraVersion("4.0.0");
        assertTrue(v1.compareTo(v2) < 0);
        assertTrue(v1.compareTo(v3) < 0);
```

I partially agree that `isLowerThan` could be useful. But it can provide conflicting results with the example you are putting, hence confusing.

```java
        // using the vars defined above.
        assertNotEquals(v1.isLowerThan(4, 0), v1.compareTo(v3) < 0);
```

Reply:
I think I should name it differently `isMajorMinorLowerThan` or have another method `trim` which would return `CassandraVersion(major + "." + minor)` - would that work better for you?

Reply:
I do not like it due to the redundancy, but can be fine with it if you insist.

Reply:
Would you be fine if we had a method like:

```java
public CassandraVersion familyLowerBound() {
    return new CassandraVersion(major, minor, 0, NO_HOTFIX, new String[0], null);
}
```
?

Reply:
That sounds good to me. It can be a constant tho.